ark/api/v1alpha1/agent_types.go

```diff
@@ -17,7 +17,7 @@ type ToolFunction struct {
 
 type AgentTool struct {
 	// +kubebuilder:validation:Required
-	// +kubebuilder:validation:Enum=built-in;custom
+	// +kubebuilder:validation:Enum=built-in;custom;agent
```

Comment:
`agent` not needed here - as long as `custom` is used then it'll handle this, apologies i know `type` is weird here we're gonna remove it